### PR TITLE
feat(docs): document how to connect CDKTF to Terraform Cloud

### LINKED ISSUE #586: Documentation on how CDK would work with Terraform Enterprise Cloud.
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Is there a recommendation on how to use terraform CDK with Terraform Enterprise cloud?

### PR DESCRIPTION
**Pull request by ansgarm**
Resolves #586

### CODE REVIEW

File: docs/working-with-cdk-for-terraform/terraform-cloud.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Connect Terraform Cloud with the CDK for Terraform
+
+This document describes two ways to connect Terraform Cloud with your CDK for Terraform codebase to continuously deploy your infrastructure.
+
+## Using the Terraform Cloud VCS Integration
+Terraform Cloud supports [connecting to VCS providers](https://www.terraform.io/docs/cloud/vcs/index.html). To use the VCS integration, commit the generated Terraform config (e.g. the `cdk.tf.json` and other files in `cdktf.out`) alongside your code so Terraform Cloud can use it to deploy your infrastructure.
```
Comment (DanielMSchmidt) [inline docs/working-with-cdk-for-terraform/terraform-cloud.md:6]: cdk.tf.json is in cdktf.out, so why not say cdktf.out?
Is there more config needed, e.g. pointing TFC to the tf.json file?

**Review by DanielMSchmidt — COMMENTED** [review]
I'd suggest showing a screenshot of the options just to be clear. Because each of the TFC options corelates with one solution here

File: docs/working-with-cdk-for-terraform/terraform-cloud.md (hunk 1 above)
Comment (ansgarm) [reply docs/working-with-cdk-for-terraform/terraform-cloud.md:6]: Yes, you're right. Going to mention that.
Also going to adjust this to just `cdktf.out` 👍

**Comment by ansgarm** [comment]
> I'd suggest showing a screenshot of the options just to be clear. Because each of the TFC options corelates with one solution here

Hm, you can use both (remote and local execution with the external CI workflow) – and VCS Integration will always do "remote execution" as it runs in TFC, so they do not correlate. Or did I misunderstood?

**Comment by DanielMSchmidt** [comment]
> I'd suggest showing a screenshot of the options just to be clear. Because each of the TFC options corelates with one solution here

You are right, I thought there is a 1:1 mapping but it's rather one only works with one and one with both, so I think it's fine like it is

**Review by DanielMSchmidt — APPROVED** [review]

**Pull request merged**